File: client/my-community/src/api/post.ts
import service from "@/utils/net"

export const addPost = (data: string, jwt: string) => {
    return service.request({ url: '/community/post/add', method: 'post', data, headers: { 'Authorization': "Bearer " + jwt } })
}

export const delPost = (data: string, jwt: string) => {
    return service.request({ url: '/community/post/del', method: 'post', data, headers: { 'Authorization': "Bearer " + jwt } })
}

export const infoPost = (data: FormData) => {
    return service.request({ url: '/community/post/info', method: 'post', data })
}

export const infoPosts = (data: FormData) => {
    return service.request({ url: '/post/info', method: 'post', data })
}

export const addComment = (data: string, jwt: string) => {
    return service.request({ url: '/post/add', method: 'post', data, headers: { 'Authorization': "Bearer " + jwt } })
}

export const appendComment = (data: string, jwt: string) => {
    return service.request({ url: '/post/append', method: 'post', data, headers: { 'Authorization': "Bearer " + jwt } })
}

export const delComment = (data: string, jwt: string) => {
    return service.request({ url: '/post/del', method: 'post', data, headers: { 'Authorization': "Bearer " + jwt } })
}

File: app/lib/api/post.dart
import 'package:dio/dio.dart';
import 'package:my_community/utils/net.dart';

Future<Response> addPost(String data, jwt) {
  return Service().getAuthService(jwt).post('/community/post/add', data: data);
}

Future<Response> delPost(String data, jwt) {
  return Service().getAuthService(jwt).post('/community/post/del', data: data);
}

Future<Response> infoPost(FormData data) {
  return Service().getService().post('/community/post/info', data: data);
}

Future<Response> infoPosts(FormData data) {
  return Service().getService().post('/post/info', data: data);
}

Future<Response> addComment(String data, jwt) {
  return Service().getAuthService(jwt).post('/post/add', data: data);
}

Future<Response> appendComment(String data, jwt) {
  return Service().getAuthService(jwt).post('/post/append', data: data);
}

Future<Response> delComment(String data, jwt) {
  return Service().getAuthService(jwt).post('/post/del', data: data);
}

Future<Response> editPost(String data, jwt) {
  return Service().getAuthService(jwt).post('/post/edit/post', data: data);
}
